tests/kafkatest/tests/streams/streams_upgrade_test.py

```diff
@@ -564,28 +563,31 @@ def do_rolling_bounce(self, processor, counter, current_generation):
                     monitors[first_other_processor] = first_other_monitor
                     monitors[second_other_processor] = second_other_monitor
 
-                    leader_monitor.wait_until("Received a future (version probing) subscription (version: 5). Sending empty assignment back (with supported version 4).",
-                                              timeout_sec=60,
-                                              err_msg="Could not detect 'version probing' attempt at leader " + str(self.leader.node.account))
+                    if counter < 3:
```

Comment:
nit: do we want to define `3` instead of leaving it as magic number?

Reply:
With the latest fixes, I think we actually don't need this any longer. We only needed if, if we get immediate rebalance. With disabling leave group request, this this should not be required any longer.